src/testdir/test_signals.vim

```diff
@@ -75,3 +77,37 @@ func Test_signal_PWR()
   bwipe!
   set updatetime& updatecount&
 endfunc
+
+" Test signal INT. Handler sets got_int. It should be like typing CTRL-C.
+func Test_signal_INT()
+  if !HasSignal('INT')
+    throw 'Skipped: INT signal not supported'
+  endif
+
+  " Skip the rest of the test when running with valgrind as signal INT is not
+  " received somehow by Vim when running with valgrind.
+  let cmd = GetVimCommand()
+  if cmd =~ 'valgrind'
+    throw 'Skipped: cannot test signal INT with valgrind'
+  endif
+
+  if !CanRunVimInTerminal()
+    throw 'Skipped: cannot run vim in terminal'
+  endif
+  let buf = RunVimInTerminal('', {'rows': 6})
+
+  " Get the PID of Vim running in terminal.
+  call term_sendkeys(buf, ":echo '<' .. getpid() .. '>'\n")
+  call WaitForAssert({-> assert_match('^<\d\+>', term_getline(buf, 6))})
+  let pid = substitute(term_getline(buf, 6), '^<\(\d\+\)>.*', '\=submatch(1)', '')
```

Comment:
You can also use term_getjob() and job_info() functions to get the
PID of the Vim instance running in the terminal.

Reply:
That's simpler indeed. Done.